src/v/cloud_storage_clients/configuration.cc

```diff
@@ -104,27 +105,38 @@ ss::future<s3_configuration> s3_configuration::make_configuration(
         }
     }
 
-    const auto endpoint_uri = [&]() -> ss::sstring {
-        if (overrides.endpoint) {
-            return overrides.endpoint.value();
-        }
-        return ssx::sformat("s3.{}.amazonaws.com", region());
-    }();
-    client_cfg.tls_sni_hostname = endpoint_uri;
+    const auto base_endpoint_uri = overrides.endpoint.value_or(
+      endpoint_url{ssx::sformat("s3.{}.amazonaws.com", region())});
```

Comment:
question: I wonder if in FIPS mode we should default to `s3-fips` rather than `s3`

Reply:
right, it seems like the natural evolution of this PR. I'll add this to a followup